isce2_topsapp/localize_slc.py

```diff
@@ -32,39 +33,43 @@ def get_session():
 
 def check_geometry(reference_obs: list,
                    secondary_obs: list,
-                   region_of_interest: list) -> GeometryCollection:
+                   frame_id: int = -1) -> GeometryCollection:
     reference_geos = [shape(r.geojson()['geometry']) for r in reference_obs]
     secondary_geos = [shape(r.geojson()['geometry']) for r in secondary_obs]
 
     reference_geo = unary_union(reference_geos)
     secondary_geo = unary_union(secondary_geos)
 
+    # if they are not Polygons they are multipolygons and not valid
+    connected_ref = (reference_geo.geom_type == 'Polygon')
+    connected_sec = (secondary_geo.geom_type == 'Polygon')
+
+    if (not connected_sec) or (not connected_ref):
+        raise RuntimeError('Reference and/or secondary dates were not connected'
+                           ' in their coverage (multipolygons)')
+
     # Two geometries must intersect for their to be an interferogram
     intersection_geo = secondary_geo.intersection(reference_geo)
     if intersection_geo.is_empty:
         raise RuntimeError('The overlap between reference and secondary scenes '
                            'is empty')
 
     # Update the area of interest based on user specification
-    if region_of_interest is not None:
-        region_of_interest_geo = box(*region_of_interest)
-        if not region_of_interest_geo.intersects(intersection_geo):
-            raise RuntimeError('Region of interest does not overlap with IFG '
-                               'area (ref and sec overlap)')
+    if frame_id != -1:
+        df_frames = gpd.read_file('s3://s1-gunw-frames/s1_frames.geojson',
```

Comment:
This file is only accessible using JPL A19 AWS credentials and won't be accessible from other deployments, or to developers locally (without effort). If you're going to keep it in an S3 bucket, it'd be best to make it a public bucket. 

However, since this file is only 72 mb, and compresses down to 22 mb with zip, I think you'd be best off just including it in the repo. That way, it's guaranteed to work for local developers, in the container, and in any deployment. Also, that way changes to the frames are explicitly  captured in PRs/git history, and documented in the changelog.

Reply:
Adding the enumerator as a dependency and reading from that would also work.

Reply:
I am going to add the enumerator as a dependency. There are issues with the frame size/overlap as it takes too long. Will get clarity soon from @dbekaert.